data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/server/ResponseHandler.java

```diff
@@ -5,27 +5,58 @@
 
 package org.opensearch.dataprepper.peerforwarder.server;
 
+import com.amazon.dataprepper.metrics.PluginMetrics;
 import com.amazon.dataprepper.model.buffer.SizeOverflowException;
 import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.common.MediaType;
+import io.micrometer.core.instrument.Counter;
 
 import java.util.Objects;
+import java.util.concurrent.TimeoutException;
 
 /**
  * Class to handle exceptions while processing HTTP POST requests by {@link PeerForwarderHttpService}
  * and sends back the HTTP response based on exception
  * @since 2.0
  */
 public class ResponseHandler {
+    static final String REQUESTS_TOO_LARGE = "requestsTooLarge";
+    static final String REQUEST_TIMEOUTS = "requestTimeouts";
+    static final String REQUESTS_UNPROCESSABLE = "requestsUnprocessable";
+    static final String BAD_REQUESTS = "badRequests";
+
+    private final Counter requestsTooLargeCounter;
+    private final Counter requestTimeoutsCounter;
+    private final Counter requestsUnprocessableCounter;
+    private final Counter badRequestsCounter;
+
+    public ResponseHandler(final PluginMetrics pluginMetrics) {
+        requestsTooLargeCounter = pluginMetrics.counter(REQUESTS_TOO_LARGE);
```

Comment:
These are nice granular metrics. But, should we also have a built-in metric which aggregates these? E.g. one for all 4xx errors?

Reply:
We have a `requestsFailedCounter` on `RemetePeerForwarder` which is count of requests whose response is either null or not OK.